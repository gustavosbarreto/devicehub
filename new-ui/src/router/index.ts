import { RouteRecordRaw, createRouter, createWebHistory } from "vue-router";
import { envVariables } from "../envVariables";
import Dashboard from "@/views/Dashboard.vue"
import Devices from "@/views/Devices.vue";
import DeviceList from "@/components/Devices/DeviceList.vue";
import DevicePendingList from "@/components/Devices/DevicePendingList.vue";
import DeviceRejectedList from "@/components/Devices/DeviceRejectedList.vue";
import DetailsDevice from "@/views/DetailsDevice.vue";
import Sessions from "@/views/Sessions.vue";
import DetailsSessions from "@/views/DetailsSessions.vue";
import FirewallRules from "@/views/FirewallRules.vue";
import PublicKeys from "@/views/PublicKeys.vue";
import Settings from "@/views/Settings.vue";
import SettingProfile from "@/components/Setting/SettingProfile.vue";
import SettingNamespace from "@/components/Setting/SettingNamespace.vue";
import SettingPrivateKeys from "@/components/Setting/SettingPrivateKeys.vue";
import SettingTags from "@/components/Setting/SettingTags.vue";
import SettingBilling from "@/components/Setting/SettingBilling.vue";


const routes: Array<RouteRecordRaw> = [
  {
    path: "/login",
    name: "login",
    component: () => import("../views/Login.vue"),
  },
  {
    path: "/forgot-pass",
    name: "ForgotPassword",
    component: () => import("../views/ForgotPassword.vue"),
  },
  {
    path: "/validation-account",
    name: "ValidationAccount",
    component: () => import("../views/ValidationAccount.vue"),
  },
  {
    path: "/update-password",
    name: "UpdatePassword",
    component: () => import("../views/UpdatePassword.vue"),
  },
  {
    path: "/sign-up",
    name: "SignUp",
    component: () => import("../views/SignUp.vue"),
  },
  {
    path: "/",
    name: "Dashboard",
    component: Dashboard,
  },
  {
    path: "/devices",
    name: "devices",
    component: Devices,
    redirect: {
      name: "listDevices",
    },
    children: [
      {
        path: "",
        name: "listDevices",
        component: DeviceList,
      },
      {
        path: "pending",
        name: "pendingDevices",
        component: DevicePendingList,
      },
      {
        path: "rejected",
        name: "rejectedDevices",
        component: DeviceRejectedList,
      },
    ],
  },
  {
    path: "/device/:id",
    name: "detailsDevice",
    component: DetailsDevice,
  },
  {
    path: "/sessions",
    name: "Sessions",
    component: Sessions,
  },
  {
    path: "/sessions/:id",
    name: "detailsSession",
    component: DetailsSessions,
  },
  {
    path: "/firewall/rules",
    name: "firewalls",
    component: FirewallRules,
  },
  {
    path: "/sshkeys/public-keys",
    name: "publicKeys",
    component: PublicKeys,
  },
  {
    path: "/settings",
    name: "settings",
    component: Settings,
    redirect: {
      name: "profileSettings",
    },
    children: [
      {
        path: "profile",
        name: "profileSettings",
        component: SettingProfile,
      },
      {
        path: "namespace-manager",
        name: "namespaceSettings",
        component: SettingNamespace,
      },
      {
        path: "private-keys",
        name: "privateKeysSettings",
        component: SettingPrivateKeys,
      },
      {
        path: "tags",
        name: "tagsSettings",
        component: SettingTags,
      },
      {
        path: "billing",
        name: "billingSettings",
        beforeEnter: (to, from, next) => {
          const enabled = envVariables.billingEnable;
          if (enabled) {
            next();
          } else {
            next("/invalid");
          }
        },
        component: SettingBilling,
      },
    ],
  },
];

const router = createRouter({
  history: createWebHistory("/"),
  routes,
});

export default router;
