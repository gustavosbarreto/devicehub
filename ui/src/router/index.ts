import { RouteRecordRaw, createRouter, createWebHistory } from "vue-router";
import { envVariables } from "../envVariables";
import { store } from "@/store";

const Dashboard = () => import("@/views/Dashboard.vue");
const Devices = () => import("@/views/Devices.vue");
const DeviceList = () => import("@/components/Devices/DeviceList.vue");
const DevicePendingList = () => import("@/components/Devices/DevicePendingList.vue");
const DeviceRejectedList = () => import("@/components/Devices/DeviceRejectedList.vue");
const Containers = () => import("@/views/Containers.vue");
const ContainerList = () => import("@/components/Containers/ContainerList.vue");
const ContainerPendingList = () => import("@/components/Containers/ContainerPendingList.vue");
const ContainerRejectedList = () => import("@/components/Containers/ContainerRejectedList.vue");
const Connectors = () => import("@/views/Connectors.vue");
const detailsConnectors = () => import("@/views/ConnectorDetails.vue");
const DetailsDevice = () => import("@/views/DetailsDevice.vue");
const Sessions = () => import("@/views/Sessions.vue");
const DetailsSessions = () => import("@/views/DetailsSessions.vue");
const FirewallRules = () => import("@/views/FirewallRules.vue");
const PublicKeys = () => import("@/views/PublicKeys.vue");
const Settings = () => import("@/views/Settings.vue");
const SettingProfile = () => import("@/components/Setting/SettingProfile.vue");
const SettingNamespace = () => import("@/components/Setting/SettingNamespace.vue");
const SettingPrivateKeys = () => import("@/components/Setting/SettingPrivateKeys.vue");
const SettingTags = () => import("@/components/Setting/SettingTags.vue");
const SettingBilling = () => import("@/components/Setting/SettingBilling.vue");

export const routes: Array<RouteRecordRaw> = [
  {
    path: "/login",
    name: "login",
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/Login.vue"),
  },
  {
    path: "/mfa-login",
    name: "MfaLogin",
    beforeEnter: (to, from, next) => {
      // Check if the user is coming from the login route
      if (from.name === "login") {
        // Allow access to MFA login if the user is coming from the login route
        next();
      } else {
        // Redirect to login if the user is not coming from the login route
        next({ name: "login" });
      }
    },
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/MfaLogin.vue"),
  },
  {
    path: "/recover-mfa",
    name: "RecoverMfa",
    beforeEnter: (to, from, next) => {
      if (from.name === "MfaLogin") {
        next();
      } else {
        next({ name: "login" });
      }
    },
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../components/AuthMFA/MfaRecover.vue"),
  },
  {
    path: "/recover-mfa/mail-sucessful",
    name: "RecoverMfaMsg",
    beforeEnter: (to, from, next) => {
      if (from.name === "RecoverMfa") {
        next();
      } else {
        next({ name: "login" });
      }
    },
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../components/AuthMFA/MfaMailRecover.vue"),
  },
  {
    path: "/forgot-pass",
    name: "ForgotPassword",
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/ForgotPassword.vue"),
  },
  {
    path: "/validation-account",
    name: "ValidationAccount",
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/ValidationAccount.vue"),
  },
  {
    path: "/reset-mfa",
    name: "MfaResetValidation",
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/MfaResetValidation.vue"),
  },
  {
    path: "/update-password",
    name: "UpdatePassword",
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/UpdatePassword.vue"),
  },
  {
    path: "/sign-up",
    name: "SignUp",
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/SignUp.vue"),
  },
  {
    path: "/confirm-account",
    name: "ConfirmAccount",
    meta: {
      layout: "LoginLayout",
      requiresAuth: false,
    },
    component: () => import("../views/ConfirmAccount.vue"),
  },
  {
    path: "/",
    name: "Dashboard",
    component: Dashboard,
  },
  {
    path: "/devices",
    name: "devices",
    component: Devices,
    redirect: {
      name: "listDevices",
    },
    children: [
      {
        path: "",
        name: "listDevices",
        component: DeviceList,
      },
      {
        path: "pending",
        name: "pendingDevices",
        component: DevicePendingList,
      },
      {
        path: "rejected",
        name: "rejectedDevices",
        component: DeviceRejectedList,
      },
    ],
  },
  {
    path: "/containers",
    name: "containers",
    beforeEnter: (to, from, next) => {
      if (!envVariables.isEnterprise && !envVariables.isCloud && envVariables.premiumPaywall) {
        store.commit("users/setShowPaywall", true);
      }
      next();
    },
    component: Containers,
    redirect: {
      name: "listContainers",
    },
    children: [
      {
        path: "",
        name: "listContainers",
        component: ContainerList,
      },
      {
        path: "pending",
        name: "pendingContainers",
        component: ContainerPendingList,
      },
      {
        path: "rejected",
        name: "rejectedContainers",
        component: ContainerRejectedList,
      },
    ],
  },
  {
    path: "/containers/connectors",
    name: "connectors",
    component: Connectors,
    beforeEnter: (to, from, next) => {
      if (!envVariables.isEnterprise && !envVariables.isCloud && envVariables.premiumPaywall) {
        store.commit("users/setShowPaywall", true);
      }
      next();
    },
  },
  {
    path: "/containers/connectors/:id",
    name: "detailsConnectors",
    component: detailsConnectors,
  },
  {
    path: "/device/:id",
    name: "detailsDevice",
    component: DetailsDevice,
  },
  {
    path: "/sessions",
    name: "Sessions",
    component: Sessions,
  },
  {
    path: "/sessions/:id",
    name: "detailsSession",
    component: DetailsSessions,
  },
  {
    path: "/firewall/rules",
    name: "firewalls",
    component: FirewallRules,
    beforeEnter: (to, from, next) => {
      if (!envVariables.isEnterprise && !envVariables.isCloud && envVariables.premiumPaywall) {
        store.commit("users/setShowPaywall", true);
      }
      next();
    },
  },
  {
    path: "/sshkeys/public-keys",
    name: "publicKeys",
    component: PublicKeys,
  },
  {
    path: "/settings",
    name: "settings",
    component: Settings,
    redirect: {
      name: "profileSettings",
    },
    children: [
      {
        path: "profile",
        name: "profileSettings",
        component: SettingProfile,
      },
      {
        path: "namespace-manager",
        name: "namespaceSettings",
        component: SettingNamespace,
      },
      {
        path: "private-keys",
        name: "privateKeysSettings",
        component: SettingPrivateKeys,
      },
      {
        path: "tags",
        name: "tagsSettings",
        component: SettingTags,
      },
      {
        path: "billing",
        name: "billingSettings",
        beforeEnter: (to, from, next) => {
          const enabled = envVariables.billingEnable;
          if (enabled) {
            next();
          } else {
            next("/404");
          }
        },
        component: SettingBilling,
      },
    ],
  },
  {
    path: "/:catchAll(.*)",
    redirect: { name: "NotFound" },
  },
  {
    path: "/404",
    name: "NotFound",
    component: () => import("../views/NotFound.vue"),
  },
];

export const router = createRouter({
  history: createWebHistory(),
  routes,
});

router.beforeEach(async (to, from, next) => {
  const isLoggedIn = store.getters["auth/isLoggedIn"];
  const layout = to.meta.layout || "AppLayout";
  const requiresAuth = to.meta.requiresAuth ?? true;

  await store.dispatch("layout/setLayout", layout);

  // Redirect to the appropriate page based on authentication status
  if (requiresAuth && !isLoggedIn) {
    next({ name: "login" }); // Redirect to login page if authentication is required and user is not logged in
  } else if (to.name === "login" && isLoggedIn) {
    next({ path: "/" }); // Redirect from login page to home if user is already logged in
  } else {
    next(); // Continue with the original navigation
  }
});
