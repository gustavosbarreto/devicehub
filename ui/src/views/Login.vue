<template>
<v-layout align-center justify-center>
  
  <v-flex xs12 sm8 md4>
    
    <div dark color="primary" class="logo">
      <div class="text-center">
        <v-icon>mdi-console</v-icon>
        <h2 style="font-family: monospace">ShellHub</h2>
        <span class="overline">beta</span>
      </div>
    </div>

    <v-card class="elevation-12">
      <v-toolbar dark color="primary">
        <v-toolbar-title>Login</v-toolbar-title>
        <v-spacer></v-spacer>
      </v-toolbar>
      
      <v-form @submit.prevent="login()" >
        <v-card-text>
          <v-text-field :error="this.$store.getters['auth/authStatus'] == 'error'" prepend-icon="person" v-model="username" label="Username" type="text"></v-text-field>
          <v-text-field :error="this.$store.getters['auth/authStatus'] == 'error'" id="password" prepend-icon="lock" v-model="password" label="Password" type="password"></v-text-field>
        </v-card-text>
        
        <v-card-actions>
          <v-spacer></v-spacer>
          <v-btn type="submit" color="primary">Submit</v-btn>
        </v-card-actions>
      </v-form>
    </v-card>

  </v-flex>
</v-layout>
</template>

<script>
export default {
  name: 'Login',

  data() {
    return {
      username: null,
      password: null
    };
  },

  methods: {
    login() {
      this.$store
        .dispatch('auth/login', {
          username: this.username,
          password: this.password
        })
        .then(() => {
          if (this.$route.query.redirect) {
            this.$router.push(this.$route.query.redirect);
          } else {
            this.$router.push('/');
          }
        });
    }
  }
};
</script>

<style lang="stylus">
.logo{
  margin-bottom: 15px;
}
</style>
