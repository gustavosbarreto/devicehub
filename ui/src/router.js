import Vue from "vue";
import Router from "vue-router";
import Dashboard from "./views/Dashboard.vue";
import store from './store'

Vue.use(Router);

const router = new Router({
    mode: "history",
    base: process.env.BASE_URL,
    routes: [
        {
            path: "/",
            name: "dashboard",
            component: Dashboard
        },
        {
            path: "/devices",
            name: "devices",

            component: () =>
                import(/* webpackChunkName: "devices" */ "./views/Devices.vue")
        },
        {
            path: "/devices/:id",
            name: "featureDevice",
            component: () =>
                import("./views/FeatureDevice.vue") 
        },
        {
            path: "/sessions",
            name: "sessions",
            component: () =>
                import("./views/Sessions.vue")
        },
        {
            path: "/login",
            name: "login",
            component: () =>
                import("./views/Login.vue")
        }
    ]
});

router.beforeEach((to, from, next) => {
    if (to.path !== '/login') {
        if (store.getters['auth/isLoggedIn']) {
          return next()
        }

        return next(`/login?redirect=${to.path}`)
      } else {
        return next()
      }
})

export default router;