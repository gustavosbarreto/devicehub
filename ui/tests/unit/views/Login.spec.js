import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Login from '@/views/Login';
import router from '@/router/index';

describe('Login', () => {
  const localVue = createLocalVue();
  localVue.use(Vuex);

  let wrapper;

  const auth = {
    status: true,
    token: 'xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx',
    user: 'user',
    tenant: 'eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoibGV',
  };

  const store = new Vuex.Store({
    state: {
      auth,
    },
    getters: {
    },
    actions: {
    },
  });

  beforeEach(() => {
    wrapper = shallowMount(Login, {
      store,
      localVue,
      router,
      mocks: {
        $route: {
          query: {},
        },
      },
    });
  });

  it('Is a Vue instance', () => {
    expect(wrapper).toBeTruthy();
  });
  it('Renders the component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('Compare data with default value', () => {
    expect(wrapper.vm.username).toEqual(null);
    expect(wrapper.vm.password).toEqual(null);
  });
  it('Renders the template with data', () => {
    const textInputUsername = wrapper.find('[data-test="username-text"]');
    const textInputPassword = wrapper.find('[data-test="password-text"]');

    expect(textInputUsername.element.value).toEqual(undefined);
    expect(textInputPassword.element.value).toEqual(undefined);

    textInputUsername.element.value = 'ShellHub';
    textInputPassword.element.value = 'ShellHub';
    expect(textInputUsername.element.value).toEqual('ShellHub');
    expect(textInputPassword.element.value).toEqual('ShellHub');

    expect(wrapper.find('[data-test="login-btn"]').exists()).toBe(true);
  });
});
