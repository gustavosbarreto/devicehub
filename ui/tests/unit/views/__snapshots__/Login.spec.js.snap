// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Renders the component 1`] = `
<v-layout-stub tag="div" align-center="" justify-center="" class="grey lighten-4">
  <v-flex-stub tag="div" xs12="" sm8="" md4="">
    <v-container-stub tag="div">
      <v-layout-stub tag="div" align-center="" justify-center="" column="">
        <v-flex-stub tag="div" class="text-center primary--text">
          <v-img-stub options="[object Object]" position="center center" src="@/assets/logo.png" transition="fade-transition"></v-img-stub> <span class="overline">BETA</span>
        </v-flex-stub>
      </v-layout-stub>
    </v-container-stub>
    <v-card-stub loaderheight="4" tag="div" color="transparent" class="elevation-0">
      <v-form-stub>
        <v-card-text-stub>
          <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" backgroundcolor="" label="Username" prependicon="person" loaderheight="2" clearicon="$clear" type="text" data-test="username-text"></v-text-field-stub>
          <v-text-field-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" backgroundcolor="" id="password" label="Password" prependicon="lock" loaderheight="2" clearicon="$clear" type="password" data-test="password-text"></v-text-field-stub>
        </v-card-text-stub>
        <v-card-actions-stub class="justify-center">
          <v-btn-stub color="primary" tag="button" activeclass="" type="submit" data-test="login-btn">
            LOGIN
          </v-btn-stub>
        </v-card-actions-stub>
        <v-card-subtitle-stub class="d-flex align-center justify-center pa-4 mx-auto">
          Don't have an account?

          <router-link-stub to="[object Object]" tag="a" ariacurrentvalue="page" event="click" class="ml-1">
            Sign up here
          </router-link-stub>
        </v-card-subtitle-stub>
      </v-form-stub>
    </v-card-stub>
  </v-flex-stub>
</v-layout-stub>
`;
